src/include/processor/result/result_set.h

```diff
@@ -4,13 +4,18 @@
 
 #include "common/data_chunk/data_chunk.h"
 #include "processor/data_pos.h"
+#include "result_set_descriptor.h"
 
 namespace kuzu {
 namespace processor {
 
-class ResultSet {
-public:
+struct ResultSet {
+    uint64_t multiplicity;
+    std::vector<std::shared_ptr<common::DataChunk>> dataChunks;
+
     explicit ResultSet(uint32_t numDataChunks) : multiplicity{1}, dataChunks(numDataChunks) {}
+    static std::unique_ptr<ResultSet> getResultSet(
```

Comment:
I think you can also make this as a constructor, which is more clear. `ResultSet::getResultSet` looks a bit confusing.